resources/server_hardware_type.rb

```diff
@@ -20,7 +20,11 @@
 
 action :edit do
   item = load_resource
-  add_or_edit(item) if item.exists?
+  if item.exists?
+    add_or_edit(item)
+  else
+    Chef::Log.warn "#{resource_name} '#{name}' does not exists"
```

Comment:
Maybe we should raise an exception. If we are trying to edit an item idempotently, the result should be the item in OneView with the same attributes passed in the recipe, and if we cannot put the item there if it does not exists then:
1. Pass through: Loses idempotent status
2. Raise exception: fails since it is an `unexpected` state